server/storage/etcd_storage.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2021 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"github.com/tikv/pd/server/kv"
+	"github.com/tikv/pd/server/storage/base"
+	"go.etcd.io/etcd/clientv3"
+)
+
+var _ Storage = (*EtcdStorage)(nil)
+
+// EtcdStorage is a storage that stores data in etcd,
+// which is used by the PD server.
+type EtcdStorage struct {
```

Comment:
Can`EtcdStoage` merge with `MemoryStorage` as `Storage`? I think etcd or memdb is the backend kv engine, and the storage is level higher than them.

Reply:
After this PR, I will try to use `EtcdStorage` and `LevelDBStorage` (maybe `BadgerDBStorage` later) to be the backend of the region storage, which will make it an independent storage component. In order to switch the backend storage smoothly, there is a need to implement different `LoadRegion`/`SaveRegion`/`DeleteRegion` according to different storage backends. For now, it's implemented in different places, the etcd implementation is in `core.Storage`, the LevelDB one is in `core.RegionStorage`. Keeping `EtcdStorage` and `LevelDBStorage` separate is a good way to make different implementations for the same method.

Reply:
This is indeed a problem, the main problem is that region storage is special... But I think a good abstraction is that I don't need to be concerned with what the underlying backend engine is.  I just know there is one storage, the storage can save/load config, rules, regions and so on. we may make region storage behavior unify for the all backend firstly, such as we can add a `WriteBatch` interface for kv.Base etc.

Reply:
The other path is what you do now, to separate them. But several concepts confuse me. As I mentioned above, ·MemoryStorage`, `EtcdStorage`,`LevelDBStroage` are different with `ConfigStorage`, `RuleStorage`... but they are all called Storage.